### PR TITLE
fix(daytona): wrap exec commands in subshell to prevent session termination

### PR DESCRIPTION
**Pull request by chaliy**
## What
Wrap user commands executed via `daytona_exec` in a subshell `( ... )` so that `exit`, SIGPIPE, SIGHUP, and other signals only terminate the subshell, not the persistent session shell.

## Why
During end-to-end testing, `daytona_exec` sessions terminated unexpectedly ~15% of the time when commands contained `bash -lc ...; exit`, pipes to `head`/`sed` (SIGPIPE), or long `&&` chains. Each failure forced expensive session re-establishment and wasted ~20% of agent budgets on retries.

Root cause: commands run directly in the persistent shell session, so any `exit` or fatal signal kills the entire session rather than just the command.

## How
- Wrap the user command in parentheses to run it in a subshell
- The shell profile preamble stays outside the subshell so PATH changes apply to the session
- Updated unit tests and integration test to match the new command format

## Risk
- Low
- Subshells are standard POSIX shell behavior. Edge case: if a command intentionally modifies the session environment, that change will not persist across exec calls. However, the shell profile preamble already re-sources environment on every exec, so this is a non-issue in practice.

## Security
- Threat categories reviewed: TM-BASH (command execution in sandbox)
- No new injection vectors - command string is unchanged, only wrapped in parentheses
- Subshells inherit the same sandbox restrictions; no privilege escalation
- No new data exposure paths

## Checklist
- [x] Tests added or updated
- [x] Backward compatibility considered
- [x] Security review performed against relevant threat model categories
- [x] All review comments addressed

Closes EVE-251

### CODE REVIEW

File: integrations/daytona/src/client.rs [hunk 1]
```diff
@@ -443,9 +443,15 @@ impl DaytonaClient {
...
+        // SIGPIPE, SIGHUP, and other signals only terminate the subshell,
+        // not the persistent session shell. Without this, commands like
+        // `bash -lc '...; exit'`, pipes to `head`/`sed` (SIGPIPE), and
+        // long `&&` chains that call `exit` would kill the entire session
+        // and force expensive re-establishment. See EVE-251.
         let cmd = match cwd {
-            Some(c) => format!("{SHELL_PROFILE_PREAMBLE}cd {c} && {command}"),
-            None => format!("{SHELL_PROFILE_PREAMBLE}{command}"),
+            Some(c) => format!("{SHELL_PROFILE_PREAMBLE}(cd {c} && {command})"),
+            None => format!("{SHELL_PROFILE_PREAMBLE}({command})"),
```
[the remark this answers is not in the document]
Comment (chaliy) [reply integrations/daytona/src/client.rs]: Good catch! Fixed in 6282a7c — added spaces inside the subshell parens to avoid `((` arithmetic context parsing.

[the remark this answers is not in the document]
Comment (chaliy) [reply integrations/daytona/src/client.rs]: Valid concern, but this is pre-existing behavior — the original code already had unquoted `cd {c}`. Keeping scope tight for this PR. The shell profile preamble and Daytona toolbox API both assume simple paths without spaces, which matches how agents use sandbox workspaces in practice.
